src/main/java/backend/resource/MultiModel.java

```diff
@@ -201,21 +202,33 @@ private void preprocessUpdatedIssues(List<Model> newModels) {
         for (Model model : newModels) {
             assert models.containsKey(model.getRepoId());
             Model existingModel = models.get(model.getRepoId());
-            if (!existingModel.getIssues().equals(model.getIssues())) {
-                // Find issues that have changed and update preferences with them
-                for (int i = 1; i <= model.getIssues().size(); i++) {
-                    // TODO O(n^2), optimise by preprocessing into a map or sorting
-                    if (!existingModel.getIssueById(i).equals(model.getIssueById(i))) {
-                        assert model.getIssueById(i).isPresent();
-                        // It's no longer currently read, but it retains its updated time.
-                        // No changes to preferences.
-                        model.getIssueById(i).get().setIsCurrentlyRead(false);
-                    }
-                }
+            if (existingModel.getIssues().equals(model.getIssues())) continue;
+            for (int i = 1; i <= model.getIssues().size(); i++) {
+                if (existingModel.getIssueById(i).equals(model.getIssueById(i)) || 
+                    !wasNotJustMarkedAsRead(model.getIssueById(i).get())) continue;
```

Comment:
The double negation is quite confusing. Could we negate the method name instead?